engine/src/test/java/io/zeebe/engine/processor/workflow/deployment/CreateDeploymentMultiplePartitionsTest.java

```diff
@@ -247,6 +249,94 @@ public void shouldIncrementWorkflowVersions() {
         .containsOnly(2);
   }
 
+  @Test
+  public void shouldFilterDuplicateWorkflow() {
+    // given
+    final Record<DeploymentRecordValue> original =
+        ENGINE.deployment().withXmlResource("process.bpmn", WORKFLOW).deploy();
+
+    // when
+    final Record<DeploymentRecordValue> repeated =
+        ENGINE.deployment().withXmlResource("process.bpmn", WORKFLOW).deploy();
+
+    // then
+    assertThat(repeated.getKey()).isGreaterThan(original.getKey());
+
+    final List<DeployedWorkflow> originalWorkflows = original.getValue().getDeployedWorkflows();
+    final List<DeployedWorkflow> repeatedWorkflows = repeated.getValue().getDeployedWorkflows();
+    assertThat(repeatedWorkflows.size()).isEqualTo(originalWorkflows.size()).isOne();
+
+    assertThat(
+            RecordingExporter.deploymentRecords(DeploymentIntent.CREATE)
+                .skip(PARTITION_COUNT)
+                .limit(PARTITION_COUNT)
+                .count())
+        .isEqualTo(PARTITION_COUNT);
+
+    final List<DeployedWorkflow> repeatedWfs =
+        RecordingExporter.deploymentRecords(DeploymentIntent.CREATED)
+            .skip(PARTITION_COUNT)
+            .limit(PARTITION_COUNT)
+            .map(r -> r.getValue().getDeployedWorkflows().get(0))
+            .collect(Collectors.toList());
+
+    assertThat(repeatedWfs.size()).isEqualTo(PARTITION_COUNT);
+    repeatedWfs.forEach(repeatedWf -> assertSameResource(originalWorkflows.get(0), repeatedWf));
+  }
+
+  @Test
+  public void shouldNotFilterDifferentWorkflows() {
+    // given
+    final Record<DeploymentRecordValue> original =
+        ENGINE.deployment().withXmlResource("process.bpmn", WORKFLOW).deploy();
+
+    // when
+    final BpmnModelInstance sameBpmnIdModel =
+        Bpmn.createExecutableProcess(PROCESS_ID).startEvent().endEvent().done();
+    final Record<DeploymentRecordValue> repeated =
+        ENGINE.deployment().withXmlResource("process.bpmn", sameBpmnIdModel).deploy();
+
+    // then
+    final List<DeployedWorkflow> originalWorkflows = original.getValue().getDeployedWorkflows();
+    final List<DeployedWorkflow> repeatedWorkflows = repeated.getValue().getDeployedWorkflows();
+    assertThat(repeatedWorkflows.size()).isEqualTo(originalWorkflows.size()).isOne();
+
+    assertDifferentResources(originalWorkflows.get(0), repeatedWorkflows.get(0));
+
+    assertThat(
+            RecordingExporter.deploymentRecords(DeploymentIntent.CREATE)
+                .skip(PARTITION_COUNT)
```

Comment:
Hey @MiguelPires sorry for the late response, just a small question. How does this work? Why do you skip 3 records and limit to 3 ? Is the skipping included in count? :thinking:

Reply:
Hey, no problem. It skips the first 3 records which belong to the first deployment and gets the next three which should belong to the repeated deployment. Now that you point it out, it looks weird, I should be asserting that 3 records were published with the new deployment key without needing to skip the first 3.

Reply:
Aaaaah because you have 3 partitions got it :laughing: so the first 3 CREATE are skipped and then you count the other and limit to 3 because of the 3 partitions. :see_no_evil: Yes might make it more to clear if you filer by the key.

Reply:
Please fix this also in the other tests after that we can merge this PR :+1: